public/docs/_examples/lifecycle-hooks/dart/lib/after_content_component.dart

```diff
@@ -0,0 +1,115 @@
+// #docplaster
+// #docregion
+import 'package:angular2/core.dart';
+
+import 'logger_service.dart';
+
+//////////////////
+@Component(
+    selector: 'my-child',
+    template: '<input [(ngModel)]="hero">')
+class ChildComponent {
+  String hero = 'Magneta';
+}
+
+//////////////////////
+@Component(
+    selector: 'after-content',
+// #docregion template
+    template: '''
+    <div>-- projected content begins --</div>
+      <ng-content></ng-content>
+    <div>-- projected content ends --</div>
+    <p *ngIf="comment != null" class="comment">{{comment}}</p>
+    '''
+// #enddocregion template
+    )
+// #docregion hooks
+class AfterContentComponent implements AfterContentChecked, AfterContentInit {
+  String _prevHero = '';
+  String comment = '';
+
+  // Query for a CONTENT child of type `ChildComponent`
+  @ContentChild(ChildComponent) ChildComponent contentChild;
+
+// #enddocregion hooks
+  LoggerService _logger;
+  AfterContentComponent(this._logger) {
+    _logIt('AfterContent constructor');
+  }
+
+// #docregion hooks
+  ngAfterContentInit() {
+    // contentChild is set after the content has been initialized
+    _logIt('AfterContentInit');
+    _doSomething();
+  }
+
+  ngAfterContentChecked() {
+    // contentChild is updated after the content has been checked
+    if (_prevHero == contentChild?.hero) {
+      _logIt('AfterContentChecked (no change)');
+    } else {
+      _prevHero = contentChild?.hero;
+      _logIt('AfterContentChecked');
+      _doSomething();
+    }
+  }
+// #enddocregion hooks
+// #docregion do-something
+  // This surrogate for real business logic sets the `comment`
+  _doSomething() {
+    comment = contentChild.hero.length > 10 ? "That's a long name" : '';
+  }
+// #enddocregion do-something
+
+  void _logIt(String method) {
+    var child = contentChild;
+    var message = "${method}: ${child != null ? child.hero:'no'} child content";
+    _logger.log(message);
+  }
+// #docregion hooks
+  // ...
```

Comment:
The `...`? It is used in the prose to indicate that there is more in that class but it is not being shown. That convention is used on the TS side.